pom.xml

```diff
@@ -147,9 +149,37 @@
 		<dependency>
 			<groupId>org.apache.maven.shared</groupId>
 			<artifactId>maven-invoker</artifactId>
-    		<version>3.0.1</version>
+			<version>3.0.1</version>
 		</dependency>
-	</dependencies>
+		<dependency>
+			<groupId>org.junit.platform</groupId>
+			<artifactId>junit-platform-launcher</artifactId>
+			<version>1.6.0</version>
+		</dependency>
+
+		<!-- 
+			Runtime dependencies to inject Test Engines
+			to be used by the UnifiedTestRunner.
+		 -->
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter-engine</artifactId>
+			<version>${junit.jupiter.version}</version>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.junit.vintage</groupId>
+			<artifactId>junit-vintage-engine</artifactId>
+			<version>${junit.jupiter.version}</version>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>com.github.testng-team</groupId>
+			<artifactId>testng-junit5</artifactId>
+			<version>0.0.1</version>
+			<scope>runtime</scope>
+		</dependency>
```

Comment:
I was looking at this again, and I am not sure if we should support TestNG at the moment through the unified platform. It seems that this tool is not officially supported by TestNG, so I am curious how this will continue to be developed in the future.

Reply:
If we just remove this runtime dependency, we should be able to by-pass the Unified Engine? The other alternative is to wrap all this crap in try-blocks and try each applicable test type (unified, junit3, junit4, testng) and take the one that works correctly the first time around.

Reply:
fabd838ccc5edffb72731718311fd111656eb70f adds necessary exception handling early in the test-type-recognition phase to be able to avoid the kind of exceptions that you caught. Additionally, i took the trouble of adding other try-catch blocks so that when running Tacoco as part of an experiment, it does not take the whole experiment's Python/Bash script down with it.

Reply:
the try-catch  in containsExecutableTest (as part of fabd838ccc5edffb72731718311fd111656eb70f) will ensure that it returns `false` if there is any issue in correctly discovering a test in a given testClass. I might even add a "dry run" mechanism to try running a discovered test from the testClass to see if the discovery actually discovered an executable test, or if it just discovered something that seemed like a test but really it was nothing.

Reply:
I went ahead and setup the basic bits for an optional test dry run in 3b59e09bca6ac6d0df6e651857acefad3f4bbd4f. We can do the necessary plumbing to make sure that this is configurable via command line options or Tacoco's config file.